opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -175,7 +176,7 @@ namespace Opm {
         /// \param[in, out] reservoir_state   reservoir state variables
         /// \param[in, out] well_state        well state variables
         template <class NonlinearSolverType>
-        IterationReport nonlinearIteration(const int iteration,
+        SimulatorReport nonlinearIteration(const int iteration,
```

Comment:
IMO this method is unfortunate because it basically duplicates the functionality of the report (and always has been) which will lead to inconsistencies or at least confusion.

maybe the report should be stored as an attributed and accessed via a `lastReport()` member?